### PR TITLE
fix(components): [time-picker] fix the panel don't show

### LINKED ISSUE #9842: [Component] [date-picker] DatePicker需要点击两次才能打开选择框
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.31`
- Element Plus Version: `2.2.0`
- Browser / OS: `Chrome 99 / MacOS 11`
- Build Tool: `Vite`

### Related Component

- `el-date-picker`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwiZGVtby1kYXRldGltZS1waWNrZXJcIj5cbiAgICA8ZGl2IGNsYXNzPVwiYmxvY2tcIj5cbiAgICAgIDxzcGFuIGNsYXNzPVwiZGVtb25zdHJhdGlvblwiPkRlZmF1bHQ8L3NwYW4+XG4gICAgICA8ZWwtZGF0ZS1waWNrZXJcbiAgICAgICAgdi1tb2RlbD1cInZhbHVlMVwiXG4gICAgICAgIHR5cGU9XCJkYXRldGltZVwiXG4gICAgICAgIHBsYWNlaG9sZGVyPVwiU2VsZWN0IGRhdGUgYW5kIHRpbWVcIlxuICAgICAgLz5cbiAgICA8L2Rpdj5cbiAgICA8ZGl2IGNsYXNzPVwiYmxvY2tcIj5cbiAgICAgIDxzcGFuIGNsYXNzPVwiZGVtb25zdHJhdGlvblwiPldpdGggc2hvcnRjdXRzPC9zcGFuPlxuICAgICAgPGVsLWRhdGUtcGlja2VyXG4gICAgICAgIHYtbW9kZWw9XCJ2YWx1ZTJcIlxuICAgICAgICB0eXBlPVwiZGF0ZXRpbWVcIlxuICAgICAgICBwbGFjZWhvbGRlcj1cIlNlbGVjdCBkYXRlIGFuZCB0aW1lXCJcbiAgICAgICAgOnNob3J0Y3V0cz1cInNob3J0Y3V0c1wiXG4gICAgICAvPlxuICAgIDwvZGl2PlxuICAgIDxkaXYgY2xhc3M9XCJibG9ja1wiPlxuICAgICAgPHNwYW4gY2xhc3M9XCJkZW1vbnN0cmF0aW9uXCI+V2l0aCBkZWZhdWx0IHRpbWU8L3NwYW4+XG4gICAgICA8ZWwtZGF0ZS1waWNrZXJcbiAgICAgICAgdi1tb2RlbD1cInZhbHVlM1wiXG4gICAgICAgIHR5cGU9XCJkYXRldGltZVwiXG4gICAgICAgIHBsYWNlaG9sZGVyPVwiU2VsZWN0IGRhdGUgYW5kIHRpbWVcIlxuICAgICAgICA6ZGVmYXVsdC10aW1lPVwiZGVmYXVsdFRpbWVcIlxuICAgICAgLz5cbiAgICA8L2Rpdj5cbiAgICA8YnIvPlxuICAgIDxkaXY+XG4gICAgICA8ZWwtaW5wdXQgdi1tb2RlbD1cInZhbHVlNFwiIC8+XG4gICAgICBcbiAgICA8L2Rpdj5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgcmVmIH0gZnJvbSAndnVlJ1xuXG5jb25zdCB2YWx1ZTEgPSByZWYoJycpXG5jb25zdCB2YWx1ZTIgPSByZWYoJycpXG5jb25zdCB2YWx1ZTMgPSByZWYoJycpXG5jb25zdCB2YWx1ZTQgPSByZWYoJycpXG5jb25zdCBkZWZhdWx0VGltZSA9IG5ldyBEYXRlKDIwMDAsIDEsIDEsIDEyLCAwLCAwKVxuXG5jb25zdCBzaG9ydGN1dHMgPSBbXG4gIHtcbiAgICB0ZXh0OiAnVG9kYXknLFxuICAgIHZhbHVlOiBuZXcgRGF0ZSgpLFxuICB9LFxuICB7XG4gICAgdGV4dDogJ1llc3RlcmRheScsXG4gICAgdmFsdWU6ICgpID0+IHtcbiAgICAgIGNvbnN0IGRhdGUgPSBuZXcgRGF0ZSgpXG4gICAgICBkYXRlLnNldFRpbWUoZGF0ZS5nZXRUaW1lKCkgLSAzNjAwICogMTAwMCAqIDI0KVxuICAgICAgcmV0dXJuIGRhdGVcbiAgICB9LFxuICB9LFxuICB7XG4gICAgdGV4dDogJ0Egd2VlayBhZ28nLFxuICAgIHZhbHVlOiAoKSA9PiB7XG4gICAgICBjb25zdCBkYXRlID0gbmV3IERhdGUoKVxuICAgICAgZGF0ZS5zZXRUaW1lKGRhdGUuZ2V0VGltZSgpIC0gMzYwMCAqIDEwMDAgKiAyNCAqIDcpXG4gICAgICByZXR1cm4gZGF0ZVxuICAgIH0sXG4gIH0sXG5dXG48L3NjcmlwdD5cbjxzdHlsZSBzY29wZWQ+XG4uZGVtby1kYXRldGltZS1waWNrZXIge1xuICBkaXNwbGF5OiBmbGV4O1xuICB3aWR0aDogMTAwJTtcbiAgcGFkZGluZzogMDtcbiAgZmxleC13cmFwOiB3cmFwO1xufVxuLmRlbW8tZGF0ZXRpbWUtcGlja2VyIC5ibG9jayB7XG4gIHBhZGRpbmc6IDMwcHggMDtcbiAgdGV4dC1hbGlnbjogY2VudGVyO1xuICBib3JkZXItcmlnaHQ6IHNvbGlkIDFweCB2YXIoLS1lbC1ib3JkZXItY29sb3IpO1xuICBmbGV4OiAxO1xufVxuLmRlbW8tZGF0ZXRpbWUtcGlja2VyIC5ibG9jazpsYXN0LWNoaWxkIHtcbiAgYm9yZGVyLXJpZ2h0OiBub25lO1xufVxuLmRlbW8tZGF0ZXRpbWUtcGlja2VyIC5kZW1vbnN0cmF0aW9uIHtcbiAgZGlzcGxheTogYmxvY2s7XG4gIGNvbG9yOiB2YXIoLS1lbC10ZXh0LWNvbG9yLXNlY29uZGFyeSk7XG4gIGZvbnQtc2l6ZTogMTRweDtcbiAgbWFyZ2luLWJvdHRvbTogMjBweDtcbn1cbjwvc3R5bGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

两个问题：
第一个：
1）点击第一个日期输入框，弹窗正常打开
2）点击第二个日期输入框，弹窗不能打开
3）再次点击第二个日期输入框，弹窗才能正常打开
第二个：
1）点击输入框，获取焦点
2）点击第一个日期输入框，弹窗不能打开
3）再次点击第一个日期输入框，弹窗才能正常打开

### What is Expected?

弹窗正常打开

### What is actually happening?

弹窗不能正常打开

### Additional comments

const handleFocusInput = (e?: FocusEvent) => {
  if (
    props.readonly ||
    pickerDisabled.value ||
    pickerVisible.value ||
    ignoreFocusEvent
  ) {
    return
  }
  // 这里进行isNil判断出问题了
  pickerVisible.value = isNil(e?.relatedTarget)
  // 改成以下方式，可以解决出现的问题，但是不知道是否有其他影响
  // pickerVisible.value = isRangeInput.value ? isNil(e?.relatedTarget) : true
  emit('focus', e)
}

### PR DESCRIPTION
**Pull request by btea**
Please make sure these boxes are checked before submitting your PR, thank you!

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

fix #9842

### CODE REVIEW

**Comment by chenxch** [comment]
[repo](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuICBpbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5zZXR1cEVsZW1lbnRQbHVzKClcblxuY29uc3QgdmFsdWUxID0gcmVmKCcnKVxuY29uc3QgdmFsdWUyID0gcmVmKCcnKVxuY29uc3QgdmFsdWUzID0gcmVmKCcnKVxuY29uc3QgdmFsdWU0ID0gcmVmKCcnKVxuY29uc3QgZGVmYXVsdFRpbWUgPSBuZXcgRGF0ZSgyMDAwLCAxLCAxLCAxMiwgMCwgMClcblxuY29uc3Qgc2hvcnRjdXRzID0gW1xuICB7XG4gICAgdGV4dDogJ1RvZGF5JyxcbiAgICB2YWx1ZTogbmV3IERhdGUoKSxcbiAgfSxcbiAge1xuICAgIHRleHQ6ICdZZXN0ZXJkYXknLFxuICAgIHZhbHVlOiAoKSA9PiB7XG4gICAgICBjb25zdCBkYXRlID0gbmV3IERhdGUoKVxuICAgICAgZGF0ZS5zZXRUaW1lKGRhdGUuZ2V0VGltZSgpIC0gMzYwMCAqIDEwMDAgKiAyNClcbiAgICAgIHJldHVybiBkYXRlXG4gICAgfSxcbiAgfSxcbiAge1xuICAgIHRleHQ6ICdBIHdlZWsgYWdvJyxcbiAgICB2YWx1ZTogKCkgPT4ge1xuICAgICAgY29uc3QgZGF0ZSA9IG5ldyBEYXRlKClcbiAgICAgIGRhdGUuc2V0VGltZShkYXRlLmdldFRpbWUoKSAtIDM2MDAgKiAxMDAwICogMjQgKiA3KVxuICAgICAgcmV0dXJuIGRhdGVcbiAgICB9LFxuICB9LFxuXVxuPC9zY3JpcHQ+XG48c3R5bGUgc2NvcGVkPlxuLmRlbW8tZGF0ZXRpbWUtcGlja2VyIHtcbiAgZGlzcGxheTogZmxleDtcbiAgd2lkdGg6IDEwMCU7XG4gIHBhZGRpbmc6IDA7XG4gIGZsZXgtd3JhcDogd3JhcDtcbn1cbi5kZW1vLWRhdGV0aW1lLXBpY2tlciAuYmxvY2sge1xuICBwYWRkaW5nOiAzMHB4IDA7XG4gIHRleHQtYWxpZ246IGNlbnRlcjtcbiAgYm9yZGVyLXJpZ2h0OiBzb2xpZCAxcHggdmFyKC0tZWwtYm9yZGVyLWNvbG9yKTtcbiAgZmxleDogMTtcbn1cbi5kZW1vLWRhdGV0aW1lLXBpY2tlciAuYmxvY2s6bGFzdC1jaGlsZCB7XG4gIGJvcmRlci1yaWdodDogbm9uZTtcbn1cbi5kZW1vLWRhdGV0aW1lLXBpY2tlciAuZGVtb25zdHJhdGlvbiB7XG4gIGRpc3BsYXk6IGJsb2NrO1xuICBjb2xvcjogdmFyKC0tZWwtdGV4dC1jb2xvci1zZWNvbmRhcnkpO1xuICBmb250LXNpemU6IDE0cHg7XG4gIG1hcmdpbi1ib3R0b206IDIwcHg7XG59XG48L3N0eWxlPlxuPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwiZGVtby1kYXRldGltZS1waWNrZXJcIj5cbiAgICA8ZGl2IGNsYXNzPVwiYmxvY2tcIj5cbiAgICAgIDxzcGFuIGNsYXNzPVwiZGVtb25zdHJhdGlvblwiPkRlZmF1bHQ8L3NwYW4+XG4gICAgICA8ZWwtZGF0ZS1waWNrZXJcbiAgICAgICAgdi1tb2RlbD1cInZhbHVlMVwiXG4gICAgICAgIHR5cGU9XCJkYXRldGltZVwiXG4gICAgICAgIHBsYWNlaG9sZGVyPVwiU2VsZWN0IGRhdGUgYW5kIHRpbWVcIlxuICAgICAgLz5cbiAgICA8L2Rpdj5cbiAgICA8ZGl2IGNsYXNzPVwiYmxvY2tcIj5cbiAgICAgIDxzcGFuIGNsYXNzPVwiZGVtb25zdHJhdGlvblwiPldpdGggc2hvcnRjdXRzPC9zcGFuPlxuICAgICAgPGVsLWRhdGUtcGlja2VyXG4gICAgICAgIHYtbW9kZWw9XCJ2YWx1ZTJcIlxuICAgICAgICB0eXBlPVwiZGF0ZXRpbWVcIlxuICAgICAgICBwbGFjZWhvbGRlcj1cIlNlbGVjdCBkYXRlIGFuZCB0aW1lXCJcbiAgICAgICAgOnNob3J0Y3V0cz1cInNob3J0Y3V0c1wiXG4gICAgICAvPlxuICAgIDwvZGl2PlxuICAgIDxkaXYgY2xhc3M9XCJibG9ja1wiPlxuICAgICAgPHNwYW4gY2xhc3M9XCJkZW1vbnN0cmF0aW9uXCI+V2l0aCBkZWZhdWx0IHRpbWU8L3NwYW4+XG4gICAgICA8ZWwtZGF0ZS1waWNrZXJcbiAgICAgICAgdi1tb2RlbD1cInZhbHVlM1wiXG4gICAgICAgIHR5cGU9XCJkYXRldGltZVwiXG4gICAgICAgIHBsYWNlaG9sZGVyPVwiU2VsZWN0IGRhdGUgYW5kIHRpbWVcIlxuICAgICAgICA6ZGVmYXVsdC10aW1lPVwiZGVmYXVsdFRpbWVcIlxuICAgICAgLz5cbiAgICA8L2Rpdj5cbiAgICA8YnIvPlxuICAgIDxkaXY+XG4gICAgICA8ZWwtaW5wdXQgdi1tb2RlbD1cInZhbHVlNFwiIC8+XG4gICAgICBcbiAgICA8L2Rpdj5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTk4NTItZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiXG4gIH1cbn0iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy05ODUyLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICBsaW5rLmhyZWYgPSAnaHR0cHM6Ly9wcmV2aWV3LTk4NTItZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gIH0pXG59IiwiX28iOnsic2hvd0hpZGRlbiI6dHJ1ZSwic3R5bGVTb3VyY2UiOiJodHRwczovL3ByZXZpZXctOTg1Mi1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyJ9fQ==)
focus
It doesn't have the right focus.

**Comment by lait233** [comment]
希望快点合并发版。不然这个问题还是挺麻烦的

**Comment by btea** [comment]
> [repo](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuICBpbXBvcnQgeyByZWYgfSBmcm9tICd2dWUnXG5zZXR1cEVsZW1lbnRQbHVzKClcblxuY29uc3QgdmFsdWUxID0gcmVmKCcnKVxuY29uc3QgdmFsdWUyID0gcmVmKCcnKVxuY29uc3QgdmFsdWUzID0gcmVmKCcnKVxuY29uc3QgdmFsdWU0ID0gcmVmKCcnKVxuY29uc3QgZGVmYXVsdFRpbWUgPSBuZXcgRGF0ZSgyMDAwLCAxLCAxLCAxMiwgMCwgMClcblxuY29uc3Qgc2hvcnRjdXRzID0gW1xuICB7XG4gICAgdGV4dDogJ1RvZGF5JyxcbiAgICB2YWx1ZTogbmV3IERhdGUoKSxcbiAgfSxcbiAge1xuICAgIHRleHQ6ICdZZXN0ZXJkYXknLFxuICAgIHZhbHVlOiAoKSA9PiB7XG4gICAgICBjb25zdCBkYXRlID0gbmV3IERhdGUoKVxuICAgICAgZGF0ZS5zZXRUaW1lKGRhdGUuZ2V0VGltZSgpIC0gMzYwMCAqIDEwMDAgKiAyNClcbiAgICAgIHJldHVybiBkYXRlXG4gICAgfSxcbiAgfSxcbiAge1xuICAgIHRleHQ6ICdBIHdlZWsgYWdvJyxcbiAgICB2YWx1ZTogKCkgPT4ge1xuICAgICAgY29uc3QgZGF0ZSA9IG5ldyBEYXRlKClcbiAgICAgIGRhdGUuc2V0VGltZShkYXRlLmdldFRpbWUoKSAtIDM2MDAgKiAxMDAwICogMjQgKiA3KVxuICAgICAgcmV0dXJuIGRhdGVcbiAgICB9LFxuICB9LFxuXVxuPC9zY3JpcHQ+XG48c3R5bGUgc2NvcGVkPlxuLmRlbW8tZGF0ZXRpbWUtcGlja2VyIHtcbiAgZGlzcGxheTogZmxleDtcbiAgd2lkdGg6IDEwMCU7XG4gIHBhZGRpbmc6IDA7XG4gIGZsZXgtd3JhcDogd3JhcDtcbn1cbi5kZW1vLWRhdGV0aW1lLXBpY2tlciAuYmxvY2sge1xuICBwYWRkaW5nOiAzMHB4IDA7XG4gIHRleHQtYWxpZ246IGNlbnRlcjtcbiAgYm9yZGVyLXJpZ2h0OiBzb2xpZCAxcHggdmFyKC0tZWwtYm9yZGVyLWNvbG9yKTtcbiAgZmxleDogMTtcbn1cbi5kZW1vLWRhdGV0aW1lLXBpY2tlciAuYmxvY2s6bGFzdC1jaGlsZCB7XG4gIGJvcmRlci1yaWdodDogbm9uZTtcbn1cbi5kZW1vLWRhdGV0aW1lLXBpY2tlciAuZGVtb25zdHJhdGlvbiB7XG4gIGRpc3BsYXk6IGJsb2NrO1xuICBjb2xvcjogdmFyKC0tZWwtdGV4dC1jb2xvci1zZWNvbmRhcnkpO1xuICBmb250LXNpemU6IDE0cHg7XG4gIG1hcmdpbi1ib3R0b206IDIwcHg7XG59XG48L3N0eWxlPlxuPHRlbXBsYXRlPlxuICA8ZGl2IGNsYXNzPVwiZGVtby1kYXRldGltZS1waWNrZXJcIj5cbiAgICA8ZGl2IGNsYXNzPVwiYmxvY2tcIj5cbiAgICAgIDxzcGFuIGNsYXNzPVwiZGVtb25zdHJhdGlvblwiPkRlZmF1bHQ8L3NwYW4+XG4gICAgICA8ZWwtZGF0ZS1waWNrZXJcbiAgICAgICAgdi1tb2RlbD1cInZhbHVlMVwiXG4gICAgICAgIHR5cGU9XCJkYXRldGltZVwiXG4gICAgICAgIHBsYWNlaG9sZGVyPVwiU2VsZWN0IGRhdGUgYW5kIHRpbWVcIlxuICAgICAgLz5cbiAgICA8L2Rpdj5cbiAgICA8ZGl2IGNsYXNzPVwiYmxvY2tcIj5cbiAgICAgIDxzcGFuIGNsYXNzPVwiZGVtb25zdHJhdGlvblwiPldpdGggc2hvcnRjdXRzPC9zcGFuPlxuICAgICAgPGVsLWRhdGUtcGlja2VyXG4gICAgICAgIHYtbW9kZWw9XCJ2YWx1ZTJcIlxuICAgICAgICB0eXBlPVwiZGF0ZXRpbWVcIlxuICAgICAgICBwbGFjZWhvbGRlcj1cIlNlbGVjdCBkYXRlIGFuZCB0aW1lXCJcbiAgICAgICAgOnNob3J0Y3V0cz1cInNob3J0Y3V0c1wiXG4gICAgICAvPlxuICAgIDwvZGl2PlxuICAgIDxkaXYgY2xhc3M9XCJibG9ja1wiPlxuICAgICAgPHNwYW4gY2xhc3M9XCJkZW1vbnN0cmF0aW9uXCI+V2l0aCBkZWZhdWx0IHRpbWU8L3NwYW4+XG4gICAgICA8ZWwtZGF0ZS1waWNrZXJcbiAgICAgICAgdi1tb2RlbD1cInZhbHVlM1wiXG4gICAgICAgIHR5cGU9XCJkYXRldGltZVwiXG4gICAgICAgIHBsYWNlaG9sZGVyPVwiU2VsZWN0IGRhdGUgYW5kIHRpbWVcIlxuICAgICAgICA6ZGVmYXVsdC10aW1lPVwiZGVmYXVsdFRpbWVcIlxuICAgICAgLz5cbiAgICA8L2Rpdj5cbiAgICA8YnIvPlxuICAgIDxkaXY+XG4gICAgICA8ZWwtaW5wdXQgdi1tb2RlbD1cInZhbHVlNFwiIC8+XG4gICAgICBcbiAgICA8L2Rpdj5cbiAgPC9kaXY+XG48L3RlbXBsYXRlPlxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0X21hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9wcmV2aWV3LTk4NTItZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiXG4gIH1cbn0iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy05ODUyLWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJlbGVtZW50LXBsdXMuanMiOiJpbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5pbXBvcnQgRWxlbWVudFBsdXMgZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5sZXQgaW5zdGFsbGVkID0gZmFsc2VcbmF3YWl0IGxvYWRTdHlsZSgpXG5cbmV4cG9ydCBmdW5jdGlvbiBzZXR1cEVsZW1lbnRQbHVzKCkge1xuICBpZiAoaW5zdGFsbGVkKSByZXR1cm5cbiAgY29uc3QgaW5zdGFuY2UgPSBnZXRDdXJyZW50SW5zdGFuY2UoKVxuICBpbnN0YW5jZS5hcHBDb250ZXh0LmFwcC51c2UoRWxlbWVudFBsdXMpXG4gIGluc3RhbGxlZCA9IHRydWVcbn1cblxuZXhwb3J0IGZ1bmN0aW9uIGxvYWRTdHlsZSgpIHtcbiAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICBsaW5rLmhyZWYgPSAnaHR0cHM6Ly9wcmV2aWV3LTk4NTItZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5jc3MnXG4gICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gIH0pXG59IiwiX28iOnsic2hvd0hpZGRlbiI6dHJ1ZSwic3R5bGVTb3VyY2UiOiJodHRwczovL3ByZXZpZXctOTg1Mi1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyJ9fQ==) focus [ focus ](https://user-images.githubusercontent.com/23251408/191538896-479bc1f8-2fb4-4b5f-8061-18bae74ed079.gif) [ ](https://user-images.githubusercontent.com/23251408/191538896-479bc1f8-2fb4-4b5f-8061-18bae74ed079.gif) It doesn't have the right focus.

Thanks for the reminder, this is indeed a issue.

**Comment by btea** [comment]
@Giwayume Hi, could you help me review this pr?

File: packages/components/focus-trap/src/focus-trap.vue [hunk 1]
```diff
@@ -175,7 +175,11 @@ export default defineComponent({
           // Give embedded focus layer time to pause this layer before reclaiming focus
           // And only reclaim focus if it should currently be trapping
           setTimeout(() => {
-            if (!focusLayer.paused && props.trapped) {
+            if (
+              !focusLayer.paused &&
+              props.trapped &&
+              trapContainer.contains(document.activeElement)
```
Comment (Giwayume) [inline packages/components/focus-trap/src/focus-trap.vue:181]: So, the intent of this design was to allow the consuming component of the focus trap to control whether or not focus should be trapped by using `props.trapped`. This code is basically negating that functionality. You could definitely end up breaking a ton of other components with this.

If you want the trap to auto-escape on tab, do something like this:

```
<template>
    <focus-trap :trapped="isTrapped" @focusout-prevented="onFocusOutTrap" />
</template>
<script>
const isTrapped = ref(false);

function onFocusOutTrap() {
    isTrapped.value = false;
}
</script>

```

Comment (Giwayume) [reply packages/components/focus-trap/src/focus-trap.vue:181]: Alright

Comment (Giwayume) [reply packages/components/focus-trap/src/focus-trap.vue:181]: This is the problematic line:

https://github.com/opengraphica/element-plus/blob/dev/packages/components/time-picker/src/common/picker.vue#L370

It was previously `pickerVisible.value = true`

`e?.relatedTarget` isn't going to exist the first time you click on the picker. I'm trying to find when that change was made to see why it was done.

**Comment by Giwayume** [comment]
Oh, it's caused by the last change to that file. https://github.com/element-plus/element-plus/issues/9460

**Comment by btea** [comment]
Yes, I added the logic below which seems to fix the problem.
```ts
pickerVisible.value = isRangeInput.value ? isNil(e?.relatedTarget) : true
```
But it has the following issue
https://github.com/element-plus/element-plus/pull/9852#issuecomment-1253834788

**Comment by Giwayume** [comment]
@btea See https://github.com/element-plus/element-plus/pull/9941/files

This should fix the original issue where the change event wasn't firing when changing end date the correct way, without causing the popup to no longer open.

**Pull request closed without merging**